torch/csrc/distributed/rpc/message.cpp

```diff
@@ -16,8 +16,13 @@ Message::Message(
     std::vector<char>&& payload,
     std::vector<torch::Tensor>&& tensors,
     MessageType type,
-    int64_t id)
-    : payload_(payload), tensors_(tensors), type_(type), id_(id) {}
+    int64_t id,
+    const AutogradMetadata& autograd_metadata)
+    : payload_(payload),
+      tensors_(tensors),
+      type_(type),
+      id_(id),
+      autograd_metadata_(autograd_metadata) {}
```

Comment:
This and the following are all distributed autograd specific, and won't be relevant for all other types of messages. I think we should create a new message type for it and add these features into that specific message type (like script_call, script_ret). 

I agree with @satgera  RpcAgent implementations should not be aware of autograd informations.